examples/infra/mod.rs

```diff
@@ -0,0 +1,657 @@
+        implementations::{
+            CentralizedCommChannel, CentralizedServerNetwork, Libp2pCommChannel, Libp2pNetwork,
+            MemoryStorage,
+        },
+        NetworkError, NodeImplementation, Storage,
+    },
+    types::{HotShotHandle, SignatureKey},
+    HotShot, ViewRunner,
+};
+use hotshot_centralized_server::{
+    FromServer, NetworkConfig, Run, RunResults, TcpStreamUtil, TcpStreamUtilWithRecv,
+    TcpStreamUtilWithSend, ToServer,
+};
+use hotshot_types::{
+    data::{TestableLeaf, ValidatingLeaf, ValidatingProposal},
+    traits::{
+        election::Election,
+        metrics::NoMetrics,
+        network::CommunicationChannel,
+        node_implementation::NodeType,
+        state::{TestableBlock, TestableState},
+    },
+    HotShotConfig,
+};
+use libp2p::{
+    identity::{
+        ed25519::{Keypair as EdKeypair, SecretKey},
+        Keypair,
+    },
+    multiaddr::{self, Protocol},
+    Multiaddr, PeerId,
+};
+use libp2p_networking::network::{MeshParams, NetworkNodeConfigBuilder, NetworkNodeType};
+#[allow(deprecated)]
+use nll::nll_todo::nll_todo;
+use tracing::{debug, error};
+
+/// yeesh maybe we should just implement SignatureKey for this...
+pub fn libp2p_generate_indexed_identity(seed: [u8; 32], index: u64) -> Keypair {
+    let mut hasher = blake3::Hasher::new();
+    hasher.update(&seed);
+    hasher.update(&index.to_le_bytes());
+    let new_seed = *hasher.finalize().as_bytes();
+    let sk_bytes = SecretKey::from_bytes(new_seed).unwrap();
+    let ed_kp = <EdKeypair as From<SecretKey>>::from(sk_bytes);
+    Keypair::Ed25519(ed_kp)
+}
+
+/// libp2p helper function
+/// convert node string into multi addr
+/// node string of the form: "$IP:$PORT"
+pub fn parse_dns(s: &str) -> Result<Multiaddr, multiaddr::Error> {
+    let mut i = s.split(':');
+    let ip = i.next().ok_or(multiaddr::Error::InvalidMultiaddr)?;
+    let port = i.next().ok_or(multiaddr::Error::InvalidMultiaddr)?;
+    Multiaddr::from_str(&format!("/dns/{ip}/tcp/{port}"))
+}
+
+/// libp2p helper function
+pub fn parse_ip(s: &str) -> Result<Multiaddr, multiaddr::Error> {
+    let mut i = s.split(':');
+    let ip = i.next().ok_or(multiaddr::Error::InvalidMultiaddr)?;
+    let port = i.next().ok_or(multiaddr::Error::InvalidMultiaddr)?;
+    Multiaddr::from_str(&format!("/ip4/{ip}/tcp/{port}"))
+}
+
+pub const LIBP2P_BOOTSTRAPS_LOCAL_IPS: &[&str] = &[
+    "127.0.0.1:9100",
+    "127.0.0.1:9101",
+    "127.0.0.1:9102",
+    "127.0.0.1:9103",
+    "127.0.0.1:9104",
+    "127.0.0.1:9105",
+    "127.0.0.1:9106",
+];
+
+pub const LIBP2P_BOOTSTRAPS_REMOTE_IPS: &[&str] = &[
+    "0.ap-south-1.cluster.aws.espresso.network:9000",
+    "1.ap-south-1.cluster.aws.espresso.network:9000",
+    "0.us-east-2.cluster.aws.espresso.network:9000",
+    "1.us-east-2.cluster.aws.espresso.network:9000",
+    "2.us-east-2.cluster.aws.espresso.network:9000",
+    "0.us-west-2.cluster.aws.espresso.network:9000",
+    "1.us-west-2.cluster.aws.espresso.network:9000",
+];
```

Comment:
I think we should put these in a configuration file.  The remote IPs particularly are very likely to change, and it is nice to not have to recompile everything to update them.  My two cents 😄

Reply:
Though I could also work on doing this when I integrate the new orchestrator since I'll need to update the config stuff anyway!

Reply:
Yes, I agree! I'll try to do that in this PR (I think this should be a minor change)

Reply:
edit: perhaps not so nontrivial. [Added a separate issue for this](https://github.com/EspressoSystems/HotShot/issues/950), I'll tackle it in a separate PR.